### PR TITLE
Add `useTransaction` hook

### LINKED ISSUE #126: `useTransaction` hook
### Description

Yesterday, when trying to prepare a [Codesandbox demo](https://codesandbox.io/s/web3-ui-demo-ykvms) for `@web3-ui/hooks`, I felt that we could abstract out the `loading` and `error` state-handling for any transaction that goes through.

### Why is this needed?

The `error` & `loading` states can be better handled using a hook specifically designed to process one transaction.

### Proposed Solution or API

https://discord.com/channels/883478451850473483/920328787118419980/920328789353988148

https://gist.github.com/Dhaiwat10/7f1089acf4918c6b9ec202cd93b2b41c
https://www.diffchecker.com/5FK3pAa5

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Nazeeh21**
Closes #126 

## Description
I've added a `useTransaction` hook. This hook takes the **function** and an array of **arguments** to this function as an argument. 
This hook returns an object containing the `execute` method, `loading` state, and `error` state. On calling this `execute` method, the function passed in the hook as an argument will get executed.

### CODE REVIEW

File: packages/hooks/src/stories/UseTransaction.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { useTransaction } from '../hooks';
+
+export default {
+  title: 'Hooks/useTransaction',
+};
+
+export const Default = () => {
+  const giveAlert = (name, greeting) => {
+    alert('Hello ' + name + '! ' + greeting);
+  };
```
Comment (Dhaiwat10) [inline packages/hooks/src/stories/UseTransaction.stories.tsx:13]: This hook is meant to be for Ethereum contract calls. Can you add an example that actually calls an Ethereum contract? You can use `useContract`

Comment (Dhaiwat10) [reply packages/hooks/src/stories/UseTransaction.stories.tsx:13]: You can remove these current stories

**Review by Dhaiwat10 — CHANGES_REQUESTED** [review]

File: packages/hooks/src/stories/UseTransaction.stories.tsx (hunk 1 above)
Comment (Nazeeh21) [reply packages/hooks/src/stories/UseTransaction.stories.tsx:13]: > Sure, I'll use `useContract` and remove current stories

File: packages/hooks/src/stories/UseTransaction.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  const { connectWallet, disconnectWallet, connected } = useWallet();
+  const contract = useContract(ADDRESS, ABI);
+  const [value, setValue] = React.useState('');
+
+  const [greet, greetLoading, greetError] = useTransaction(async () =>
+    // @ts-expect-error
+    alert(await contract.greet())
+  );
+
+  // @ts-expect-error
```
Comment (Dhaiwat10) [inline packages/hooks/src/stories/UseTransaction.stories.tsx:74]: Can you paste the error that you are seeing here?

Comment (Nazeeh21) [reply packages/hooks/src/stories/UseTransaction.stories.tsx:74]: I'm getting this typescript type error `Property 'setGreeting' does not exist on type '{}'.`

image

Comment (Dhaiwat10) [reply packages/hooks/src/stories/UseTransaction.stories.tsx:74]: That's expected since the `useContract` does not return any type defs as of now

**Pull request merged**